src/_pytest/terminal.py

```diff
@@ -956,7 +956,28 @@ def show_skipped(lines):
 
 def _get_pos(config, rep):
     nodeid = config.cwd_relative_nodeid(rep.nodeid)
-    return nodeid
+    path, _, testname = nodeid.partition("::")
+    if not testname:
+        return nodeid
+
+    # Append location (line number).
+    # This uses the first traceback entry for the location in the test itself
+    # (rather than reprcrash, which might be less relevant for going to
+    # directly, e.g. pexpect failures in pytest itself).
+    try:
+        testloc = rep.longrepr.reprtraceback.reprentries[0].reprfileloc
+    except AttributeError:
+        return nodeid
+
+    if isinstance(testloc.path, six.string_types):
+        testloc_path = py.path.local(testloc.path)
+    else:
+        testloc_path = testloc.path
+    rel_testloc_path = config.invocation_dir.bestrelpath(testloc_path)
+    if rel_testloc_path == path:
+        return "%s (line %d)" % (nodeid, testloc.lineno)
+
+    return "%s (%s:%d)" % (nodeid, rel_testloc_path, testloc.lineno)
```

Comment:
In which situation does this fall into? Won't this generate very long summary lines? I fear as it is the message already takes quite all the terminal space and most of the times it gets trimmed anyway, adding this information to the summary line will make it even less useful, as more of exception message will be trimmed.

How do you feel if we include this information only when `-v` is passed?

Reply:
You mean the last case?
Ok with verbose > 0 here.
Adding only " (line XX)" is useful.

It is probably not hit anyway I would imagine (have not checked).

Reply:
About `-v`, I meant if we only added line numbers of `-v` is passed. In retrospect this could also be the case for the message summary as well I guess.

Reply:
Well, I think those are very useful to have by default and only add like ~10 extra chars.